app/src/main/java/com/simply/birthdayapp/auth/registration/data/repository/SignUpRepositoryImpl.kt

```diff
@@ -0,0 +1,57 @@
+package com.simply.birthdayapp.auth.registration.data.repository
+
+import android.util.Log
+import com.apollographql.apollo.ApolloClient
+import com.apollographql.apollo.api.ApolloResponse
+import com.simply.SignUpMutation
+import com.simply.birthdayapp.auth.registration.domain.model.SignUpInput
+import com.simply.birthdayapp.auth.registration.domain.model.User
+import com.simply.birthdayapp.auth.registration.domain.repository.SignUpRepository
+import com.simply.birthdayapp.core.result.Result
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+
+
+class SignUpRepositoryImpl(private val apolloClient: ApolloClient) : SignUpRepository {
+
+    override suspend fun signUp(signUpInput: SignUpInput): Result<User> {
+        return withContext(Dispatchers.IO) {
+            try {
+                Result.Loading<Nothing>("")
+                val response: ApolloResponse<SignUpMutation.Data> = apolloClient.mutation(
+                    SignUpMutation(
+                        com.simply.type.SignUpInput(
+                            email = signUpInput.email,
+                            firstName = signUpInput.firstName,
+                            lastName = signUpInput.lastName,
+                            password = signUpInput.password
+                        )
+                    )
+                ).execute()
+
+
+                val user = response.data?.signUp?.let {
+                    User(
+                        id = it.id,
+                        email = it.email,
+                        firstName = it.firstName,
+                        lastName = it.lastName,
+                        image = it.image
+                    )
+                }
+
+                if (user != null) {
```

Comment:
please do not use hardcode strings in repository or any layer below screen, and prevent using strings for comparing  , consider to use enums or sealed classes